src/inet/TCPEndPointImplSockets.cpp

```diff
@@ -947,10 +947,10 @@ void TCPEndPointImplSockets::ReceiveData()
         {
             VerifyOrDie(rcvLen > 0);
             size_t newDataLength = rcvBuf->DataLength() + static_cast<size_t>(rcvLen);
-            VerifyOrDie(CanCastTo<uint16_t>(newDataLength));
+            VerifyOrDie(newDataLength <= UINT32_MAX);
```

Comment:
On a 32-bit system, this check is completely meaningless, yes?

We need to figure out what we are _actually_ checking here and why.

Reply:
Removed this check from here.